components/script/dom/element.rs

```diff
@@ -534,13 +537,43 @@ impl Element {
             }
         }
 
-        // TODO: Update the following steps to align with the newer spec.
         // Step 3.
+        // If element’s local name is a valid custom element name,
+        // or element’s is value is non-null
+        if is_valid_custom_element_name(self.local_name()) || self.get_is().is_some() {
+            let definition = self.get_custom_element_definition();
+            if definition.is_some() && definition.unwrap().disable_shadow {
```

Comment:
You can use [`Option::is_some_and`](https://doc.rust-lang.org/std/option/enum.Option.html#method.is_some_and) to avoid having to call unwrap here:
```rust
definition.is_some_and(|definition| definition.disable_shadow())
```